### ISSUE TITLE
CRS transformations are only applied to shapefiles

### ISSUE THREAD

**Issue by NyakudyaA** [qgis/QGIS#38066]
# Problem

* Download the shapefile attached here.
* Use the Data source manager to load the layer.
* A CRS transformation dialog pop us up like below
![CRS](https://user-images.githubusercontent.com/2510900/89010006-14314e80-d30e-11ea-88c2-4bc9f6e3ec5c.png)
* Select the CRS transformation and the layer gets loaded into QGIS.

**Second procedure**
* Navigate to the file system and load the vector layer into PostGIS using ogr2ogr/shp2pgsql specifying the CRS 26332
```
for file in `ls *.shp`;do shp2pgsql -s 26332 -c -D -I -W LATIN1  $file  | psql  -d foo;done
```
* Use the data source manager/ QGIS browser to load the  PostGIS layer into QGIS. 
* Notice that the layer is loaded without any CRS transformation requested.

[building_footprint.zip](https://github.com/qgis/QGIS/files/5005796/building_footprint.zip)

**Settings**
My CRS settings are set to ask for datum transformations if available
![settings-crs](https://user-images.githubusercontent.com/2510900/89013572-68d7c800-d314-11ea-81c3-06e5f6ad3fe5.png)

**QGIS Version**
![QGIS-Ver](https://user-images.githubusercontent.com/2510900/89013463-36c66600-d314-11ea-82d4-e2be6c7b6979.png)

**Comment by gioman** [comment]
>     * Use the data source manager/ QGIS browser to load the  PostGIS layer into QGIS.
> 
>     * Notice that the layer is loaded without any CRS transformation requested.
> 
> 
> [building_footprint.zip](https://github.com/qgis/QGIS/files/5005796/building_footprint.zip)

@NyakudyaA just tested on QGIS master on both Linux and Windows the issue is not only if the source is a shapefile or postgis but is also about how the datasource is added to the project:

Shapefile: from the QGIS browser or D&D from a file explorer or QGIS data source manager > transformation dialog shows

PostGIS: QGIS browser  > transformation dialog shows

PostGIS: DB Manager or QGIS data source manager >  > transformation dialog do **not** shows


So I guess that the title for this ticket must change to:

"transfomation dialog do not show if a PostGIS layers is added with DB Manager or data source Mananger".

**Comment by NyakudyaA** [comment]
> transfomation dialog do not show if a PostGIS layers is added with DB Manager or data source Mananger

You are right. I will change the title and when I find time I would also like to check if it's only PostGIS that is affected by this.

**Comment by Pedro-Murteira** [comment]
This is still valid on QGIS 3.22.3. The transformation dialog doesn't show via DB Manager, be it from drag-n-drop or double click.
